pkg/controller/controller_binding.go

```diff
@@ -366,12 +379,16 @@ func (c *controller) injectBinding(binding *v1alpha1.Binding, credentials map[st
 
 	found := false
 
-	_, err := c.kubeClient.Core().Secrets(binding.Namespace).Get(binding.Spec.SecretName, metav1.GetOptions{})
+	existingSecret, err := c.kubeClient.Core().Secrets(binding.Namespace).Get(binding.Spec.SecretName, metav1.GetOptions{})
 	if err == nil {
 		found = true
 	}
 
 	if found {
+		if !isOwnedByBinding(&existingSecret.ObjectMeta, binding) {
```

Comment:
I think this is an improvement for now, but I have been wondering to myself whether this is the right point to perform this check.  Say we do the check here - and cannot persist the secret with the binding result.  Does that mean that we should delete the binding?  It definitely means the binding is not ready, and implies that the binding has permanently failed.

I don't have the right answers yet, but it is something that we should think about as part of this PR.